packages/now-next/src/utils.ts

```diff
@@ -1066,6 +1075,37 @@ function isDirectory(path: string) {
   return fs.existsSync(path) && fs.lstatSync(path).isDirectory();
 }
 
+export function normalizeLocalePath(
+  pathname: string,
+  locales?: string[]
+): {
+  detectedLocale?: string;
+  pathname: string;
+} {
+  let detectedLocale: string | undefined;
+  // first item will be empty string from splitting at first char
+  const pathnameParts = pathname.split('/');
+
+  (locales || []).some(locale => {
+    if (pathnameParts[1].toLowerCase() === locale.toLowerCase()) {
+      detectedLocale = locale;
+      pathnameParts.splice(1, 1);
+      pathname = pathnameParts.join('/') || '/';
+      return true;
+    }
+    return false;
+  });
+
+  return {
+    pathname,
+    detectedLocale,
+  };
+}
+
+export function addLocale(pathname: string, locale?: string) {
+  return locale ? `/${locale}${pathname}` : pathname;
+}
```

Comment:
Maybe you could add a parameter to this function for `routesManifest` so you dont need to repeat the same fallback logic in multiple places

Reply:
Updated to de-dupe the mentioned fallback logic